server/src/main/java/org/opensearch/common/blobstore/stream/write/StreamContextSupplier.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.stream.write;
+
+import org.opensearch.common.blobstore.stream.StreamContext;
+
+/**
+ * Will return the <code>StreamContext</code> to the caller given the part size
+ */
+public interface StreamContextSupplier {
```

Comment:
FunctionalInterface?

Reply:
Yes, this is a functional interface. We have a couple of more functional interfaces, I'll add the annotation to all of them.

Reply:
do we need to address this?